src/main/kotlin/pet/project/app/service/UserService.kt

```diff
@@ -0,0 +1,39 @@
+package pet.project.app.service
+
+import mu.KotlinLogging
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import pet.project.app.exception.UserNotFoundException
+import pet.project.app.model.User
+import pet.project.app.repository.UserRepository
+
+@Service
+class UserService(private val userRepository: UserRepository) {
+
+    private val logger = KotlinLogging.logger {}
+
+    fun create(user: User): User = userRepository.save(user)
+
+    fun getById(userId: String): User {
+        return userRepository.findByIdOrNull(userId) ?: throw UserNotFoundException(userId, "GET request")
+    }
+
+    fun update(userId: String, user: User): User {
+        if (userRepository.existsById(userId)) {
+            user.id = userId
+            return userRepository.save(user)
+        }
+        throw UserNotFoundException(userId, "UPDATE request")
+    }
+
+    fun addBookToWishList(userId: String, bookId: String): Boolean {
+        TODO("Implement")
+    }
+
+    fun delete(userId: String) {
+        if (!userRepository.existsById(userId)) {
+            logger.warn { "Attempting to delete absent user with id=$userId" }
+        }
+        userRepository.deleteById(userId)
```

Comment:
Suppose we have the next case:
0. No such user `id=1` in the DB
1. `Thread 1`: invokes the function
2. `Thread 1`: checks the user `id=1` doesn't exist and logs the message
3. `Thread 2`: created the user `id=1`
4. `Thread 1`: deletes the user `id=1` (line 37)

I don't expect that the `thread 1` delete the user.


the simple fix 

```suggestion
        if (userRepository.existsById(userId)) {
            userRepository.deleteById(userId)
        } else {
            logger.warn { "Attempting to delete absent user with id=$userId" }
        }
```

The suggestion is not ideal and still has issues. But we can improve it what when we will start learning mongo

Reply:
Thx, i see the problem now. 
Fixed: Corrected code as suggested